bundles/org.eclipse.passage.lic.base/src/org/eclipse/passage/lic/base/io/FileCollection.java

```diff
@@ -49,9 +49,9 @@ public Collection<Path> get() throws LicensingException {
 		}
 	}
 
-	private Stream<Path> filesIn(Path path) throws IOException {
-		return Files.walk(path) //
-				.filter(Files::isRegularFile);
+	static Stream<Path> filesIn(Path path) throws IOException {
+		Stream<Path> paths = Files.exists(path) ? Files.walk(path) : Stream.empty();
```

Comment:
Understand. After checking the previous code again the reason for the current failure is that before the mentioned PR the `IOException` caught in `FileCollection` accidentally was not propagated as `LicensingException`. Throwing a `LicensingException` was at least the intention before. From the IOException alone it cannot be distinguished (reliably) why it occurred. So if you relly want an `IllegalArgumentException` here the existence of the start file/directory has to be checked initially.

I also checked all non-test call-sides of `FileCollection.get()`and think that `LocalConditions.licenses()` is the only caller where a non-existing start is a valid case.

Reply:
Let me put it in a simpler way: _fail safe_ is dangerous practice, Passage core has been designed on _fail fast_.

In this case not-existing path is a developer's error. It must be fixed by adding some ensutring code to `FileCollection` client.  

`FileCollection`  has clear meaning and responsibility. It must not divise answers for illegal state that can be easily avoided by caller side.

Reply:
> Let me put it in a simpler way: _fail safe_ is dangerous practice, Passage core has been designed on _fail fast_.
> 
> In this case not-existing path is a developer's error. It must be fixed by adding some ensutring code to `FileCollection` client.

I understand that and just wanted to point out, that this was not done before. But of course it can be improved. :)

The latest commit throws an `IllegalArgumentException` if the collection root does not exist.